job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/JenkinsJobManagement.java

```diff
@@ -444,13 +466,13 @@ private boolean updateExistingItem(AbstractItem item, javaposse.jobdsl.dsl.Item
         boolean created;
 
         // Leverage XMLUnit to perform diffs
-        Diff diff;
+        Diff diff = null;
         try {
             String oldJob = item.getConfigFile().asString();
             diff = XMLUnit.compareXML(oldJob, config);
             if (diff.identical()) {
                 LOGGER.log(Level.FINE, format("Item %s is identical", item.getName()));
-                notifyItemUpdated(item, dslItem);
+                // If items are identical nothing is modified
```

Comment:
notifyItemUpdated is important for the extension point and must be kept. An extension point implementation can store data outside of the config.xml and must be notified even if the config.xml has not changed. But it should not be called in a dry run.